go/simple/v1/ampli/ampli.go

```diff
@@ -0,0 +1,385 @@
+			Configuration: DefaultConfiguration,
+		}
+	}
+
+	var apiKey string
+	if options.Client.APIKey != "" {
+		apiKey = options.Client.APIKey
+	} else if options.Environment != "" {
+		apiKey = APIKey[options.Environment]
+	} else if options.Client.Configuration.APIKey != "" {
+		apiKey = options.Client.Configuration.APIKey
+	}
+
+	if !(apiKey != "" || options.Client.Instance != nil) {
+		log.Default().Printf("Error: Ampli.Load() requires option.Environment, " +
+			"and apiKey from either options.Client.APIKey or APIKey[options.Environment], " +
+			"or options.Client.Instance")
+	}
+
+	var configuration amplitude.Config
+	if !options.Client.Configuration.IsEmpty() {
+		configuration = options.Client.Configuration
+	} else {
+		configuration = DefaultConfiguration
+	}
+
+	if configuration.Plan == (amplitude.Plan{}) {
+		configuration.Plan = DefaultConfiguration.Plan
+	}
+
+	if options.Client.Instance != nil {
+		a.Client = options.Client.Instance
+	} else {
+		configuration.APIKey = apiKey
+		a.Client = amplitude.NewClient(configuration)
+	}
+}
+
+// InitializedAndEnabled checks if Ampli is initialized and enabled.
+func (a *Ampli) InitializedAndEnabled() bool {
+	return !a.Disabled
+}
+
+// Track tracks an StronglyTypedEvent.
+func (a *Ampli) Track(userID string, event StronglyTypedEvent, eventOptions amplitude.EventOptions) {
+	if !a.InitializedAndEnabled() {
+		return
+	}
+
+	if userID != "" {
+		eventOptions.UserID = userID
+	}
+
+	baseEvent := event.ToEvent()
+	baseEvent.EventOptions = eventOptions
+
+	a.Client.Track(baseEvent)
+}
+
+// Identify identifies a user and set user properties.
+func (a *Ampli) Identify(userID string, identify Identify, eventOptions amplitude.EventOptions) {
+	if !a.InitializedAndEnabled() {
+		return
+	}
+
+	if userID != "" {
+		eventOptions.UserID = userID
+	}
+
+	identifyEvent := amplitude.Event{
+		EventType: amplitude.IdentifyEventType,
+		EventProperties: map[string]interface{}{
+			"requiredNumber": identify.RequiredNumber,
+			"optionalArray":  identify.OptionalArray,
+		},
+		EventOptions: eventOptions,
+	}
+
+	a.Client.Track(identifyEvent)
+}
+
+// GroupIdentify identifies a group and set group properties.
+func (a *Ampli) GroupIdentify(groupType string, groupName string, group Group, eventOptions amplitude.EventOptions) {
+	if !a.InitializedAndEnabled() {
+		return
+	}
+
+	groupIdentifyEvent := amplitude.Event{
+		EventType: amplitude.GroupIdentifyEventType,
+		Groups:    map[string][]string{groupType: {groupName}},
+		EventProperties: map[string]interface{}{
+			"requiredBoolean": group.RequiredBoolean,
+			"optionalString":  group.OptionalString,
+		},
+		EventOptions: eventOptions,
+	}
+
+	a.Client.Track(groupIdentifyEvent)
+}
+
+// SetGroup sets group for the current user.
+func (a *Ampli) SetGroup(userID string, groupType string, groupName []string, eventOptions amplitude.EventOptions) {
+	if !a.InitializedAndEnabled() {
+		return
+	}
+
+	if eventOptions == (amplitude.EventOptions{}) {
```

Comment:
???

Reply:
Deleted this block. It doesn't make any sense here